packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/columns.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  useCurrencyFormat,
+  useColumns,
+  NonNegativeNumberInputCell,
+  ColumnAlign,
+  ExpiryDateCell,
+  CheckCell,
+  CurrencyCell,
+} from '@openmsupply-client/common';
+import { DraftOutboundLine } from '../../../types';
+
+export const useOutboundLineEditColumns = ({
+  onChange,
+}: {
+  onChange: (key: string, value: number, packSize: number) => void;
+}) => {
+  const columns = useColumns<DraftOutboundLine>(
+    [
+      [
+        'numberOfPacks',
+        {
+          Cell: NonNegativeNumberInputCell,
+          width: 100,
+          label: 'label.num-packs',
+          setter: ({ packSize, id, numberOfPacks }) =>
+            onChange(id, numberOfPacks ?? 0, packSize ?? 1),
+        },
+      ],
+      ['packSize', { width: 90 }],
+      [
+        'unitQuantity',
+        {
+          accessor: ({ rowData }) => rowData.numberOfPacks * rowData.packSize,
+          width: 90,
+        },
+      ],
+      {
+        label: 'label.available',
+        key: 'availableNumberOfPacks',
+        align: ColumnAlign.Right,
+        width: 85,
+        accessor: ({ rowData }) => rowData.stockLine?.availableNumberOfPacks,
+      },
+      {
+        label: 'label.in-store',
+        key: 'totalNumberOfPacks',
+        align: ColumnAlign.Right,
+        width: 80,
+        accessor: ({ rowData }) => rowData.stockLine?.totalNumberOfPacks,
+      },
+      'batch',
+      [
+        'expiryDate',
+        {
+          Cell: ExpiryDateCell,
+          width: 80,
+        },
+      ],
+      [
+        'locationName',
+        {
+          accessor: ({ rowData }) => rowData.location?.name,
+          width: 70,
+        },
+      ],
+      [
+        'sellPricePerPack',
+        {
+          Cell: CurrencyCell,
+          formatter: sellPrice => useCurrencyFormat(Number(sellPrice)),
+          width: 75,
+        },
+      ],
+      {
+        label: 'label.on-hold',
+        key: 'onHold',
+        Cell: CheckCell,
+        accessor: ({ rowData }) => rowData.stockLine?.onHold,
+        align: ColumnAlign.Center,
+        width: 80,
+      },
+    ],
+    {},
+    [onChange]
+  );
```

Comment:
the tax column is going to be a real stretch (cause you'd also then need a line total 🙄 )